metric_name_builder_bench_test.go

```diff
@@ -0,0 +1,120 @@
+package otlptranslator
+
+import (
+	"fmt"
+	"testing"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func BenchmarkBuildCompliantMetricName(b *testing.B) {
+	metrics := createTestScenarios()
+
+	for _, scenario := range metrics {
+		for _, withSuffixes := range []bool{true, false} {
+			b.Run(fmt.Sprintf("%s/withSuffixes=%t", scenario.name, withSuffixes), func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					BuildCompliantMetricName(scenario.metric, "test_namespace", withSuffixes)
+				}
+			})
+		}
+	}
```

Comment:
Let's use real sub-benchmarks, instead of approximating them through a slash in the name (Go separates sub-benchmarks through the slash character). Emulating a sub-benchmark through a slash in the name is at best confusing for the person running the benchmark. Maybe it otherwise acts practically the same as using a real sub-benchmark, but let's not try our luck. 

Also, can we instead have `withSuffixes` at the outer level? I'd much rather have it this way, because I can then run all cases for either `withSuffixes=true` or `withSuffixes=false`, in case there's e.g. a performance problem with adding suffixes. Performance problems with adding suffixes shouldn't be specific to the individual cases, so it's friendlier to the person benchmarking to have them a level above case sub-benchmarks:
 
```suggestion
	for _, withSuffixes := range []bool{true, false} {
		b.Run(fmt.Sprintf("withSuffixes=%t", withSuffixes), func(b *testing.B) {
			for _, scenario := range scenarios {
				b.Run(scenario.name, func(b *testing.B) {

					for i := 0; i < b.N; i++ {
						BuildCompliantMetricName(scenario.metric, "test_namespace", withSuffixes)
					}
				})
			})
		}
	}
```

Reply:
> Maybe it otherwise acts practically the same as using a real sub-benchmark, but let's not try our luck.
It does behave the same way :) 

Before committing the change with your suggestion, I could run the sub-benchmark like this:
```console
arthursens$ go test -benchmem -run=^$ -bench ^BenchmarkBuildCompliantMetricName/Basic_metric_with_no_special_characters/withSuffixes=true$ github.com/prometheus/otlptranslator
goos: darwin
goarch: arm64
pkg: github.com/prometheus/otlptranslator
cpu: Apple M2 Pro
BenchmarkBuildCompliantMetricName/Basic_metric_with_no_special_characters/withSuffixes=true-12           1399539               840.5 ns/op           246 B/op         21 allocs/op
PASS
ok      github.com/prometheus/otlptranslator    2.114s
```

No strong opinions here though, I'll adapt the code with your suggestion